graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/processor/MutationMappingModelBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.processor;
+
+import graphql.language.InputValueDefinition;
+import graphql.language.ListType;
+import graphql.language.SourceLocation;
+import graphql.language.Type;
+import graphql.language.TypeName;
+import io.stargate.graphql.schema.schemafirst.util.TypeHelper;
+import java.util.Map;
+import java.util.Optional;
+
+abstract class MutationMappingModelBuilder extends ModelBuilderBase<MutationMappingModel> {
+
+  protected MutationMappingModelBuilder(ProcessingContext context, SourceLocation location) {
+    super(context, location);
+  }
+
+  protected static EntityMappingModel findEntity(
+      InputValueDefinition input,
+      Map<String, EntityMappingModel> entities,
+      ProcessingContext context,
+      String mutationName,
+      String mutationKind)
+      throws SkipException {
+
+    Type<?> type = TypeHelper.unwrapNonNull(input.getType());
+
+    if (type instanceof ListType) {
```

Comment:
why you are checking only `ListType`? As far as I can see, there could be other types (i.e.: `MapType`).
Maybe we should check `if(!type instnanceof TypeName)`?

Reply:
No, this is `graphql.language.Type`. It represents a type reference in the AST, with only 3 possible subtypes:
* `TypeName`: a type itself, like `User`, `Boolean`
* `NonNullType`: `User!`
* `ListType`: `[User]`

Reply:
How maps are represented?

Reply:
There are no maps in GraphQL: https://spec.graphql.org/June2018/#sec-Wrapping-Types.